src/v0/destinations/am/transform.js

```diff
@@ -419,113 +541,58 @@ function responseBuilderSimple(
-          set(payload, 'adid', advertId);
-        }
-      }
-    }
-
-    payload.time = new Date(getFieldValueFromMessage(message, 'timestamp')).getTime();
-
-    // send user_id only when present, for anonymous users not required
-    if (
-      message.userId &&
-      message.userId !== '' &&
-      message.userId !== 'null' &&
-      message.userId !== null
-    ) {
-      payload.user_id = message.userId;
-    }
-    payload.session_id = getSessionId(message);
-
-    updateConfigProperty(
+    return nonAliasResponsebuilder(
       message,
       payload,
-      mappingConfig[ConfigCategory.COMMON_CONFIG.name],
-      true,
-      destination.Config,
+      destination,
+      evType,
+      groupInfo,
+      rootElementName,
     );
-
-    // we are not fixing the verson for android specifically any more because we've put a fix in iOS SDK
-    // for correct versionName
-    // ====================
-    // fixVersion(payload, message);
-
-    if (payload.user_properties) {
-      delete payload.user_properties.city;
-      delete payload.user_properties.country;
-      if (payload.user_properties.address) {
-        delete payload.user_properties.address.city;
-        delete payload.user_properties.address.country;
-      }
-    }
-
-    if (!payload.user_id && !payload.device_id) {
-      logger.debug('Either of user ID or device ID fields must be specified');
-      throw new InstrumentationError('Either of user ID or device ID fields must be specified');
-    }
-
-    payload.ip = getParsedIP(message);
-    payload.library = 'rudderstack';
-    payload = removeUndefinedAndNullValues(payload);
-    response.endpoint = endpoint;
-    response.method = defaultPostRequestConfig.requestMethod;
-    response.headers = {
-      'Content-Type': JSON_MIME_TYPE,
-    };
-    response.userId = message.anonymousId;
-    response.body.JSON = {
-      api_key: destination.Config.apiKey,
-      [rootElementName]: [payload],
-      [addOptions]: addMinIdlength(),
-    };
-    respList.push(response);
-
-    // https://developers.amplitude.com/docs/group-identify-api
-    // Refer (1.), Rudder group call updates group propertiees.
-    if (evType === EventType.GROUP && groupInfo) {
-      groupResponse.method = defaultPostRequestConfig.requestMethod;
-      groupResponse.endpoint = groupEndpoint(destination.Config);
-      let groupPayload = Object.assign(groupInfo);
-      groupResponse.userId = message.anonymousId;
-      groupPayload = removeUndefinedValues(groupPayload);
-      groupResponse.body.FORM = {
-        api_key: destination.Config.apiKey,
-        identification: [JSON.stringify(groupPayload)],
-      };
-      respList.push(groupResponse);
-    }
   }
-
   return respList;
-}
+};
 
+const getGroupInfo = (destination, groupInfo, groupTraits) => {
+  let updatedGroupInfo = { ...groupInfo };
+  const groupTypeTrait = get(destination, 'Config.groupTypeTrait');
+  const groupValueTrait = get(destination, 'Config.groupValueTrait');
```

Comment:
```suggestion
  const { groupTypeTrait, groupValueTrait } = destination.Config;
```

Reply:
Also I think we can move `let updatedGroupInfo` into the if-block

Reply:
Done